pkg/reconcilers/deployment_mutator_test.go

```diff
@@ -0,0 +1,390 @@
+			desired: func() *appsv1.Deployment {
+				d := deployment.DeepCopy()
+				d.Spec.Template.Labels = map[string]string{
+					"app": "authorino-pod-new",
+					"new": "pod-label",
+				}
+				return d
+			}(),
+			existing: func() *appsv1.Deployment {
+				d := deployment.DeepCopy()
+				d.Spec.Template.Labels = map[string]string{
+					"app":      "authorino-pod",
+					"existing": "label",
+				}
+				return d
+			}(),
+			want: true,
+			verify: func(t *testing.T, existing *appsv1.Deployment) {
+				labels := existing.Spec.Template.Labels
+				if labels["app"] != "authorino-pod" {
+					t.Errorf("expected app label not to be updated, got %s", labels["app"])
+				}
+				if labels["new"] != "pod-label" {
+					t.Error("expected new label to be added")
+				}
+				if labels["existing"] != "label" {
+					t.Error("expected existing label to be preserved")
+				}
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			desired := tt.desired.DeepCopy()
+			existing := tt.existing.DeepCopy()
+
+			got := tt.mutator(desired, existing)
+			if got != tt.want {
+				t.Errorf("expected update %v, got %v", tt.want, got)
+			}
+
+			if tt.verify != nil {
+				tt.verify(t, existing)
+			}
+		})
+	}
+}
+
+func TestDeploymentMutator(t *testing.T) {
+	t.Run("invalid object types", func(t *testing.T) {
+		mutator := DeploymentMutator(DeploymentReplicasMutator)
+
+		_, err := mutator(&corev1.Pod{}, &appsv1.Deployment{})
+		if err == nil {
+			t.Error("expected error for invalid desired type")
+		}
+
+		_, err = mutator(&appsv1.Deployment{}, &corev1.Pod{})
+		if err == nil {
+			t.Error("expected error for invalid existing type")
+		}
+	})
+
+	t.Run("multiple mutators", func(t *testing.T) {
+		desired := &appsv1.Deployment{
+			Spec: appsv1.DeploymentSpec{
+				Replicas: ptrToInt32(3),
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{Image: "new-image"},
+						},
+					},
+				},
+			},
+		}
+		existing := &appsv1.Deployment{
+			Spec: appsv1.DeploymentSpec{
+				Replicas: ptrToInt32(1),
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{Image: "old-image"},
+						},
+					},
+				},
+			},
+		}
+
+		mutator := DeploymentMutator(
+			DeploymentReplicasMutator,
+			DeploymentImageMutator,
+		)
+
+		updated, err := mutator(desired, existing)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		if !updated {
+			t.Error("expected update to be required")
+		}
+
+		if *existing.Spec.Replicas != 3 {
+			t.Errorf("expected replicas to be updated to 3, got %d", *existing.Spec.Replicas)
+		}
+		if existing.Spec.Template.Spec.Containers[0].Image != "new-image" {
+			t.Errorf("expected image to be updated to 'new-image', got %s",
+				existing.Spec.Template.Spec.Containers[0].Image)
+		}
+	})
+}
+
+func ptrToInt32(i int32) *int32 {
+	return &i
+}
```

Comment:
Do you mind using [`k8s.io/utils/ptr.To`](https://pkg.go.dev/k8s.io/utils/ptr#To) instead please?